hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -740,66 +740,23 @@ def valid_tuple_axes(ndim, min_size=0, max_size=None):
     return st.lists(axes, min_size, max_size, unique_by=lambda x: x % ndim).map(tuple)
 
 
-class BroadcastShapeStrategy(SearchStrategy):
-    def __init__(self, shape, min_dims, max_dims, min_side, max_side):
-        assert 0 <= min_side <= max_side
-        assert 0 <= min_dims <= max_dims <= 32
-        SearchStrategy.__init__(self)
-        self.shape = shape
-        self.side_strat = st.integers(min_side, max_side)
-        self.min_dims = min_dims
-        self.max_dims = max_dims
-        self.min_side = min_side
-        self.max_side = max_side
-
-    def do_draw(self, data):
-        elements = cu.many(
-            data,
-            min_size=self.min_dims,
-            max_size=self.max_dims,
-            average_size=min(
-                max(self.min_dims * 2, self.min_dims + 5),
-                0.5 * (self.min_dims + self.max_dims),
-            ),
-        )
-        result = []
-        reversed_shape = tuple(self.shape[::-1])
-        while elements.more():
-            if len(result) < len(self.shape):
-                # Shrinks towards original shape
-                if reversed_shape[len(result)] == 1:
-                    if self.min_side <= 1 and not data.draw(st.booleans()):
-                        side = 1
-                    else:
-                        side = data.draw(self.side_strat)
-                elif self.max_side >= reversed_shape[len(result)] and (
-                    not self.min_side <= 1 <= self.max_side or data.draw(st.booleans())
-                ):
-                    side = reversed_shape[len(result)]
-                else:
-                    side = 1
-            else:
-                side = data.draw(self.side_strat)
-            result.append(side)
-        assert self.min_dims <= len(result) <= self.max_dims
-        assert all(self.min_side <= s <= self.max_side for s in result)
-        return tuple(reversed(result))
-
-
 @st.defines_strategy
 def broadcastable_shapes(shape, min_dims=0, max_dims=None, min_side=1, max_side=None):
     # type: (Shape, int, int, int, int) -> st.SearchStrategy[Shape]
     """Return a strategy for generating shapes that are broadcast-compatible
     with the provided shape.
 
+    This strategy simply calls the ``multiple_shapes`` strategy with ``inputs=1``,
+    and returns the sole generated shape.
+
```

Comment:
I think it's worth keeping this as a separate strategy - while it's obviously implemented in terms of the more complex strategy, it's also independently useful at a much lower level of Numpy knowledge.

IMO `multiple_shapes` has sufficiently different uses that we should not explicitly document this link - either it's obvious to the user, or they don't need to know.

Reply:
Agreed!